### PR TITLE
Add flag to remove containers on commit

### PR DESCRIPTION
**Pull request by rhatdan**
I think this would be good practice to eliminate wasted disk space.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
I first brought this up back in https://github.com/projectatomic/buildah/issues/2

File: cmd/buildah/commit.go [hunk 1]
```diff
@@ -128,5 +132,8 @@ func commitCmd(c *cli.Context) error {
 		return errors.Wrapf(err, "error committing container %q to %q", builder.Container, image)
 	}
 
+	if c.IsSet("reference-time") {
```
Comment (nalind) [inline cmd/buildah/commit.go]: Copy-paste error?

File: tests/basic.bats [hunk 2]
```diff
@@ -95,6 +95,12 @@ load helpers
   cmp ${TESTDIR}/other-randomfile $yetanothernewroot/other-randomfile
   buildah delete $yetanothernewcid
 
+  newcid=$(buildah from new-image)
+  buildah commit --rm --signature-policy ${TESTSDIR}/policy.json $newcid containers-storage:remove-container-image
+  newroot=$(buildah mount $newcid)
+  [ "$status" -ne 0 ]
```
Comment (nalind) [inline tests/basic.bats]: Unless I'm mistaken, `bats(7)` says that `$status` is only updated when you use its `run` helper, so I don't think this'd be checking the exit status of the right command.  The `run` helper also captures the output of the command as `$output`, so there's no need for a subshell here.

File: docs/buildah-commit.md [hunk 3]
```diff
@@ -33,11 +33,14 @@ Control the format for the image manifest and configuration data.  Recognized
 formats include *oci* (OCI image-spec v1.0, the default) and *docker* (version
 2, using schema format 2 for the manifest).
 
+**--rm**
+Remove the container content after committing it to an image.  Default leaves the container content.
```
Comment (TomSweeneyRedHat) [inline docs/buildah-commit.md]: I'd suggest:  "Remove the container and its content after committing it to an image.  Default leaves the container and its content in place.

File: cmd/buildah/commit.go (hunk 1 above)
Comment (TomSweeneyRedHat) [inline cmd/buildah/commit.go]: 👍

Comment (rhatdan) [inline cmd/buildah/commit.go]: Fixed

File: tests/basic.bats (hunk 2 above)
Comment (rhatdan) [inline tests/basic.bats]: Fixed

File: docs/buildah-commit.md (hunk 3 above)
Comment (rhatdan) [inline docs/buildah-commit.md]: Sounds good.

**Comment by rhatdan** [comment]
@nalind @TomSweeneyRedHat PTAL

**Comment by nalind** [comment]
LGTM.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+ f03bfb1

**Comment by nalind** [comment]
@rh-atomic-bot retry

**Pull request closed without merging**